File: api/src/Domain/Settings/Entities/ServiceConfiguration.cs
using System;

namespace DetailingArsenal.Domain {
    public class ServiceConfiguration : Entity<Service> {
        public Guid ServiceId { get; set; }
        public Guid? VehicleCategoryId { get; set; }

        /// <summary>
        /// The price of the service in USD.
        /// </summary>
        public decimal Price { get; set; }

        /// <summary>
        /// Duration in minutes.
        /// </summary>
        public int Duration {
            get => duration;
            set {
                if (value < 0) {
                    throw new ArgumentOutOfRangeException();
                }

                duration = value;
            }
        }

        private int duration;

        public static ServiceConfiguration Create(Guid serviceId, Guid? vehicleCategoryId, decimal price, int duration) {
            return new ServiceConfiguration() {
                Id = Guid.NewGuid(),
                ServiceId = serviceId,
                VehicleCategoryId = vehicleCategoryId,
                Price = price,
                Duration = duration
            };
        }
    }
}

File: app/src/api/scheduling/settings/data-transfer-objects/service-configuration.ts
/**
 * Configuration that retains a price and time duration for a service.
 */
export class ServiceConfiguration {
    public static PRICE_MIN_VALUE = 0;
    public static DURATION_MIN_VALUE = 0;

    /**
     * Create a new service config.
     * @param vehicleCategoryId The ID of the vehicle category it applies to.
     * @param price The price in USD.
     * @param duration The amount of minutes it takes.
     */
    constructor(public id: string, public vehicleCategoryId: string, public price: number, public duration: number) {
        if (this.duration < ServiceConfiguration.DURATION_MIN_VALUE) {
            throw new RangeError('duration');
        }

        if (this.price < ServiceConfiguration.PRICE_MIN_VALUE) {
            throw new RangeError('price');
        }
    }
}
